### PR TITLE
CMake: Use targets for dependencies instead of variables

### PR DESCRIPTION
**Pull request by Nopey**
- Added a new script, cmake-scripts/Dependencies.cmake
- Dependencies.cmake creates cmake targets for each depenency, eg. openomf::argtable, that contain all the necessary link libraries, include paths, and compiler defines (including openomf-specific ones like ARGTABLE3_FOUND)
- Depenendencies.cmake knows to use find_package CONFIG mode in vcpkg builds, and find_package module mode in other builds.
- vcpkg shims have been removed from FindXYZ.cmake scripts, as they are no longer necessary.
- optional dependencies (CUnit, miniupnpc, natpmp) are now only built by vcpkg when a corresponding option is set (USE_TESTS, USE_MINIUPNPC, and USE_NATPMP).
- miniupnpc and natpmp are now only linked to when a corresponding option is set: USE_MINIUPNPC and USE_NATPMP
- Requesting an optional dep (eg USE_MINIUPNPC) will cause the build to fail if the dep cannot be found; this prevents miniupnpc and natpmp from silently regressing in CI.

Closes #708, as this PR contains it.

### CODE REVIEW

**Comment by Nopey** [comment]
Rebased on latest master and squashed into two commits-- one for Vagabond's libnatpmp port (from PR 708) and one for the cmake changes.

File: cmake-scripts/Dependencies.cmake
Comment (katajakasa) [inline cmake-scripts/Dependencies.cmake:11]: Could these if-elses be minimized to just contain stuff that is different ? And move the identical lines outside of the if-else block ?

**Review by katajakasa — CHANGES_REQUESTED** [review]

File: cmake-scripts/Dependencies.cmake
Comment (Nopey) [reply cmake-scripts/Dependencies.cmake:11]: Moved the add_library calls outside of the vcpkg_toolchain if/else's, thanks-- that helps make it clear what targets Dependencies.cmake is providing to the rest of the scripts.
Note that I kept the find_package calls intentionally within the if-else blocks.

Also fixed the mess of tabs and spaces that was the indentation in this file (oops)

**Review by katajakasa — APPROVED** [review]
LGTM as far as I know about vcpkg. Although I am kind of annoyed how much additional stuff it seems to be requiring.

**Comment by Nopey** [comment]
rebased after cmakelists.txt conflict, gonna wait for CI to pass before a merge

**Comment by Nopey** [comment]
borked it heh

**Comment by Nopey** [comment]
removed platform.h.in and check_symbol_exists(strdup) stuff from CMakeLists.txt, since previous pr already pulled out it out.

**Review by katajakasa — APPROVED** [review]

**Comment by Nopey** [comment]
CI passed and pre-rebase version had approval-- merging.

**Pull request merged**
